frontend/src/components/StatsCard.js

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from "react";
+import { Stack, Text, chakra, Box, SimpleGrid, Link } from "@chakra-ui/react";
+import { TriangleDownIcon, TriangleUpIcon } from "@chakra-ui/icons";
+import useNFTs from "../core/hooks/useNFTs";
+import web3 from "web3";
+
+const TIME_PERIOD = {
+  current: 0,
+  previous: 1,
+};
+
+const isNumberNotZero = (str) => {
+  if (isNaN(Number(str) || Number(str) == 0)) {
+    return false;
+  } else {
+    return true;
+  }
+};
+
+const getEthValue = (string) => {
+  const ether = web3.utils.fromWei(string, "ether");
+  return nFormatter(ether, 3);
+};
+
+const nFormatter = (num, digits) => {
+  const lookup = [
+    { value: 1, symbol: "" },
+    { value: 1e3, symbol: "k" },
+    { value: 1e6, symbol: "M" },
+    { value: 1e9, symbol: "G" },
+    { value: 1e12, symbol: "T" },
+    { value: 1e15, symbol: "P" },
+    { value: 1e18, symbol: "E" },
+  ];
+  const rx = /\.0+$|(\.[0-9]*[1-9])0+$/;
+  var item = lookup
+    .slice()
+    .reverse()
+    .find(function (item) {
+      return num >= item.value;
+    });
+  return item
+    ? (num / item.value).toFixed(digits).replace(rx, "$1") + item.symbol
+    : "0";
+};
+
+const getChange = (a, b) => {
+  if (isNumberNotZero(a) && isNumberNotZero(b)) {
+    let retval = (Math.abs(Number(a) - Number(b)) * 100) / Number(b);
+    retval = retval > 9999 ? nFormatter(retval, 3) : retval;
```

Comment:
What if `retval` is negative? Maybe you should use `Math.abs()`?

Reply:
Good point

Reply:
Also removed .toFixed from return as this can break in case when retval > 9999 is true and nFormatter returns a string. 

```js
retval =
      Math.abs(retval) > 9999 ? nFormatter(retval, 2) : retval.toFixed(2);
    return retval;
```